lib/base-compiler.js

```diff
@@ -172,12 +179,27 @@ Compile.prototype.compile = function (source, options, filters) {
                         result.asm = "<Compilation failed>";
                         return result;
                     }
+                    result.hasOptOutput = false;
+                    if(self.couldHaveOptOutput(options, self.compiler.version)) {
+                        const optPath = path.join(dirPath, outputFilebase + ".opt.yaml");
+                        if(fs.existsSync(optPath)) {
```

Comment:
Ideally I'd use another promise here to avoid the sync, but in practice I can't see how it'd cause any issues to be synchronous.